Diplom_2 - my work.

AlenaTerletskaya opened the pull request:
В pom.xml подключены библиотеки:  junit, gson, rest-assured, allure-junit4, allure-rest-assured, aspectjweaver, lombok, javafaker.

 В pom.xml добавлены плагины: maven-surefire-plugin, allure-maven.
 Написаны тесты:
 по созданию пользователя  - пакет user.create_user: UserCreateTest, UserCreateWithoutRequiredFieldTest;
 по логину пользователя - пакет user.login: UserLoginTest, UserLoginInvalidEmailOrPasswordTest;
 по изменению пользователя - пакет user.change_user: AuthorizedUserChangeTest, UnauthorizedUserChangeTest, UserChangeWithExistingEmailTest;
 по созданию заказов - пакет order.create_order: OrderCreateTest, OrderCreateWithoutTokenTest;
 по получению списка заказов - пакет order: GetOrdersTest.
 Созданы клиенты для отправки запросов к эндпойнтам в пакете clients: Client, UserClient, OrderClient.
 Созданы модели данных:
 в пакете models.user: User, Credentials, UserLogout;
 в пакете models.order: Order;
 в пакете models.order.ingredients: Ingredient, Ingredients.
 Созданы генераторы данных в пакете data: UserGenerator, OrderGenerator.
 Добавлен отчет allure.

AlenaTerletskaya committed:
В pom.xml подключены библиотеки:  junit, gson, rest-assured, allure-junit4, allure-rest-assured, aspectjweaver, lombok, javafaker.
 В pom.xml добавлены плагины: maven-surefire-plugin, allure-maven.
 Написаны тесты:
 по созданию пользователя  - пакет user.create_user: UserCreateTest, UserCreateWithoutRequiredFieldTest;
 по логину пользователя - пакет user.login: UserLoginTest, UserLoginInvalidEmailOrPasswordTest;
 по изменению пользователя - пакет user.change_user: AuthorizedUserChangeTest, UnauthorizedUserChangeTest, UserChangeWithExistingEmailTest;
 по созданию заказов - пакет order.create_order: OrderCreateTest, OrderCreateWithoutTokenTest;
 по получению списка заказов - пакет order: GetOrdersTest.
 Созданы клиенты для отправки запросов к эндпойнтам в пакете clients: Client, UserClient, OrderClient.
 Созданы модели данных:
 в пакете models.user: User, Credentials, UserLogout;
 в пакете models.order: Order;
 в пакете models.order.ingredients: Ingredient, Ingredients.
 Созданы генераторы данных в пакете data: UserGenerator, OrderGenerator.
 Добавлен отчет allure.

nikolay-977 commented on src/main/java/data/OrderGenerator.java:
```diff
@@ -0,0 +1,145 @@
+
+// Класс для генерации данных заказа
+public class OrderGenerator {
+
```
Необходимо исправить: убери лишние пропуски строк. Для зрительного разделения кода достаточно использовать один пропуск строки. Между фигурными скобками и после аннотаций пропуски не нужны. Пропуски строк должны быть однообразными.

nikolay-977 commented on src/main/java/data/OrderGenerator.java:
```diff
@@ -0,0 +1,145 @@
+        Ingredients ingredients = orderClient.getIngredients();
+        String invalidIngr = "a" + ingredients.getData().get(0).get_id().substring(1);
+
+//        Faker faker = new Faker();
```
Необходимо исправить: удали лишний комментированный код

nikolay-977 commented on src/test/java/order/create_order/OrderCreateTest.java:
```diff
@@ -0,0 +1,168 @@
+        // Авторизуемся под пользователем. Проверяем, что авторизация прошла успешно.
+        ValidatableResponse responseLogin = userClient.loginUser(Credentials.from(user));
+        accessToken = responseLogin.extract().path("accessToken");
+        Assert.assertEquals("Login status code should be equal to " + SC_OK,
```
Необходимо исправить: Проверки должны быть в конце тестовых методов. Между проверками и после проверок не должно быть дополнительных действий. Иначе в случае фейла первой проверки, последующие действия(шаги и методы) не будут выполнены, это может аффектить другие тесты, что увеличит время локализации дефекта. Если нужны дополнительные проверки, лучше реализовать дополнительные кейсы.

nikolay-977 commented on src/test/java/user/change_user/AuthorizedUserChangeTest.java:
```diff
@@ -0,0 +1,101 @@
+    }
+
+    // Тестовые данные
+    @Parameterized.Parameters
```
Можно улучшить: В параметризованных тестах для аннотации @Parameterized.Parameters лучше использовать аргумент name: @Parameterized.Parameters(name = "Тестовые данные: {0} {1}"), где {0}, {1} - индексы параметров. Это повысит информативность проверки.

AlenaTerletskaya committed:
Во всех классах убрала лишние пропуски строк, сделала их однообразными.
Убрала лишний комментированный код в классе OrderClient,
В тестовых классах вынесла проверки в конец тестовых методов.
Убрала неиспользуемые переменные в методах:
- GetOrdersTest.unauthUserCanNotGetHisOrders - переменную responseLogout.
- AuthorizedUserChangeTest.authorizedUserCanBeChanged - переменную responseCreate.
Добавила к аннотации @Parameterized.Parameters аргумент name в классах:
- AuthorizedUserChangeTest,
- UnauthorizedUserChangeTest,
- UserCreateWithoutRequiredFieldTest,
- UserLoginInvalidEmailOrPasswordTest.
